src/libcore/iter/mod.rs

```diff
@@ -643,6 +643,43 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
             _ => (usize::MAX, None)
         }
     }
+
+    #[inline]
+    fn all<F>(&mut self, f: F) -> bool where F: FnMut(Self::Item) -> bool {
+        self.iter.clone().chain(self.orig.clone()).all(f)
+    }
+
+    #[inline]
+    fn max(self) -> Option<Self::Item> where Self::Item: cmp::Ord { self.orig.clone().max() }
+
+    #[inline]
+    fn min(self) -> Option<Self::Item> where Self::Item: cmp::Ord {
+        cmp::min(self.iter.min(), self.orig.clone().min())
```

Comment:
`self.iter` is a subset of `self.orig`, so the minimum of `self.orig` will be at least as small as the minimum of `self.iter`.

Reply:
Yeah — this is necessary to preserve the consistency of the behaviour of `min` when multiple elements are equally minimum. It's necessary to return the first/leftmost one, which in this case may not be the same as the first one in `self.orig`.